framework/src/play/utils/ErrorsCookieCrypter.java

```diff
@@ -0,0 +1,22 @@
+package play.utils;
+
+import play.exceptions.UnexpectedException;
+import play.libs.Crypter;
+
+import javax.annotation.Nullable;
+
+public class ErrorsCookieCrypter {
+  private final Crypter crypter = new Crypter("errors-");
+
+  public String encrypt(String errorsCookie) {
+    return crypter.encryptAES(errorsCookie);
```

Comment:
Please add some dynamic content to errorsCookie (like `Math.random()` or `System.nanoTime()`, otherwise hacker can see both encrypted and decrypted error message, and can calculate play secret key.

Reply:
improved